Add RichMenu tutorial

HellPie opened the pull request:
With #52 fixes #37 completely.

The tutorial follows a similar flow and should be a bit better than the one initially pushed with #52.

bdistin commented:
Lots of duplication, and this should be a child of the RichDisplay tutorial. See #37 for more details on what this tutorial should include.

HellPie commented:
Oh so you actually want a single tutorial markdown doc for both?

bdistin commented:
See like: https://github.com/dirigeants/klasa/blob/master/tutorials/names.json#L42 It creates a section at the bottom that says Further Reading. With the appropriate links. The child page then would only include information on how it extends the parent/what's the same, whats different/and examples without repeating a lot of what's already written/the same in the parent class.

HellPie commented:
I can't seem to be able to wake up GitHub but I pushed both requested changes, the new tutorial is in #804be72b310da32eab8c0dbc57922a66b2d54276.

I'm hoping it'll update this pull request soon enough (?)

bdistin commented on tutorials/RichMenu.md:
```diff
@@ -0,0 +1,78 @@
+While {@tutorial RichDisplay} allows you to create fully configurable paginated embeds, {@link RichMenu} allows you to define menus using a similar interface.
+
+Unlike {@tutorial RichDisplay}, {@link RichMenu} doesn't allow to define how pages will look and instead of calling {@link RichDisplay.addPage} that allows to customize the template (if provided), the user is presented with the {@link RichMenu.addOption} method, which only requires a `name` and a `body` and handles the organization of the options automatically.
```
>doesn't allow to define how pages will

could use better wording

HellPie replied:
manages a page's content and layout automatically?

bdistin commented on tutorials/RichMenu.md:
````diff
@@ -0,0 +1,78 @@
+
+```javascript
+return collector.message.edit(new this.client.methods.Embed()
+	.setDescription(`The chosen command is \`${command.name}\`.`)
````
Why not expand this and show full command details (description/extended help/usage/ect).

HellPie replied:
I really only wanted to give a simple practical example (based on yours from the RichMenu PR) that showcased how it worked.
I feel like it would be wrong to enphasize on what could be defined normal javascript (I access a class, I do "stuff" with the class) instead of keeping the RichMenu at the center of the tutorial and avoiding everything extra that does not pertain to it.

bdistin replied:
The example as it stands is too lazy for the quality of tutorial this should be. This requested change will not be dismissed without further address.

HellPie commented:
I started working on a full rewrite of this tutorial as well as a better version of the RichDisplay tutorial with more advanced examples, but I'm busy at work, so this pull request won't see new stuff for as long as a month from now.

I'm really sorry, I'm trying my best to pull off 3 projects, these 2 prs and work, but I'm really just too busy with the latter as of now.

HellPie commented:
Addressed all issues, I didn't change opinion on how useful the changes are.

If there is anything else to change I'll happily do it during the weekend if I'm gonna end up feeling better, otherwise it's gonna be a while until I can get back to this again for some serious improvements.
